h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -431,7 +437,9 @@ public void compute2() {
             rs._tsk.join();
           } else if(rs.isMutating()) // c) => throw IAE
             throw new IllegalArgumentException("Can not compute rollup stats while vec is being modified. (3)");
-        } else { // d) => compute the rollups
+        } else { // rs == null d) => compute the rollups
+          final Vec vec = DKV.getGet(_vecKey);
+          if (vec == null) break; // Vec was deleted since we last checked
```

Comment:
Also: if we fail due to an exception because of deleted Vec after we acquire the lock then we will leak the key for rollups (with flag "computing").